light-system-programs/programs/verifier_program_two/src/lib.rs

```diff
@@ -26,10 +30,19 @@ pub enum ErrorCode {
     InvalidVerifier,
 }
 
+#[derive(Clone)]
+pub struct TransactionConfig;
+impl Config for TransactionConfig {
+    /// Number of nullifiers to be inserted with the transaction.
+    const NR_NULLIFIERS: usize = 4;
+    /// Number of output utxos.
+    const NR_LEAVES: usize = 4;
+    /// ProgramId.
+    const ID: Pubkey = pubkey!("2cxC8e8uNYLcymH6RTGuJs3N8fXGkwmMpw45pY65Ay86");
```

Comment:
I think we can eliminate the TransactionConfig struct now, since:
- we have const generics for NR_NULLIFIERS & NR_LEAVES
- the context struct has the program_id field [see docs](https://docs.rs/anchor-lang/0.28.0/anchor_lang/context/struct.Context.html#structfield.program_id)

Reply:
if we don't remove the struct let's remove the ID field for sure and use the ctx.program_id

Reply:
Good point, will do

Reply:
I couldn't eliminate it fully - we still need to inject program ID here

https://github.com/Lightprotocol/light-protocol/blob/a0394abae45f128a6e86048ea8743cb084c70aa9/light-verifier-sdk/src/state.rs#L55-L61

Reply:
hm couldn't we do a const generic for the id in this case?

Reply:
No - `Pubkey` cannot be passed directly as a const generic, it's a non-primitive tyoe.

Passing `TransactionConfig` works as a hack, because what we pass is an empty struct and the fields are smuggled in a trait.

Reply:
```
error: `str` is forbidden as the type of a const generic parameter
  --> src/state.rs:44:56
   |
44 | impl<const NR_CHECKED_INPUTS: usize, const PROGRAM_ID: str> anchor_lang::AccountSerialize
   |                                                        ^^^
   |
   = note: the only supported types are integers, `bool` and `char`
   = help: more complex types are supported with `#![feature(adt_const_params)]`
```

so only primitives can be used as const generics.